src/WsReconnect/index.ts

```diff
@@ -0,0 +1,97 @@
+import TimeNode from '../TimeNode';
+import { ReconnectMsg } from '../Enum';
+import W3Util from '../Util';
+
+declare const setTimeout: any;
+
+export default class WsReconnect {
+  private timenode: TimeNode;
+  private reconnectTries: number = 0;
+  private reconnecting: boolean = false;
+  private reconnected: boolean = false;
+
+  constructor(timenode: TimeNode) {
```

Comment:
I would decouple TimeNode from WsReconnect entirely. I think WsReconnect is a good example to be event emitter -> https://ourcodeworld.com/articles/read/445/how-to-use-event-emitters-with-es5-and-es6-in-node-js-easily instead of calling TimeNode directly.

Also I think adding simple IProvider interface and put it directly in ctor would be better.

Reply:
Okay :+1: